src/test/java/com/artipie/rpm/RpmTest.java

```diff
@@ -44,58 +49,94 @@
 final class RpmTest {
 
     /**
-     * Path of repomd.xml fil.
+     * Abc test rmp file.
      */
-    private static final String REPOMD = "repodata/repomd.xml";
+    private static final String ABC =
+        "src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm";
+
+    /**
+     * Libdeflt test rmp file.
+     */
+    private static final String LIBDEFLT =
+        "src/test/resources-binary/libdeflt1_0-2020.03.27-25.1.armv7hl.rpm";
 
     @Test
-    void doesntBrakeMetadataWhenInvalidPackageSent()
-        throws Exception {
-        final Storage storage = new InMemoryStorage();
-        final Rpm repo =  new Rpm(
-            storage, StandardNamingPolicy.SHA1, Digest.SHA256, true
-        );
-        storage.save(
-            new Key.From("oldfile.rpm"),
-            new Content.From(
-                Files.readAllBytes(
-                    Paths.get("src/test/resources-binary/abc-1.01-26.git20200127.fc32.ppc64le.rpm")
-                )
-            )
-        );
+    void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp) throws Exception {
+        final Storage storage = new FileStorage(tmp);
+        final Rpm repo =  new Rpm(storage, StandardNamingPolicy.SHA1, Digest.SHA256, true);
+        RpmTest.addRpm(storage, "oldfile.rpm", RpmTest.ABC);
         repo.batchUpdate(Key.ROOT).blockingAwait();
-        final String repomd = new String(
-            new Concatenation(
-                storage.value(new Key.From(RpmTest.REPOMD)).get()
-            ).single().blockingGet().array(),
-            Charset.defaultCharset()
-        );
         final byte[] broken = {0x00, 0x01, 0x02 };
-        storage.save(
-            new Key.From("broken.rpm"),
-            new Content.From(
-                broken
-            )
+        storage.save(new Key.From("broken.rpm"), new Content.From(broken)).get();
+        RpmTest.addRpm(storage, "new.rpm", RpmTest.LIBDEFLT);
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        MatcherAssert.assertThat(
+            countData(tmp),
```

Comment:
@olenagerasimova I would be better if you assert the exact expected metadata file, not just a fragment of it.